docs/v0.9/en/user-guide/operations/compaction.md

```diff
@@ -0,0 +1,134 @@
+
+For example, during the first compaction, the time span for all SST files is 4 hours, then GreptimeDB will choose 12 hours as the time window for that table and persist it for later compactions.
+
+GreptimeDB considers the window contains most recently inserted timestamps as **active window** and those previous windows as **inactive windows**.
+
+### Sorted runs
+Sorted runs is a collection of SST files that have sorted and non-overlapping time ranges. 
+
+For example, a table contains 5 SSTs with following time ranges (all inclusive): `[0, 10]`, `[12, 23]`, `[22, 25]`,`[24, 30]`,`[26,33]` and we can find 2 sorted runs:
+
+![num-of-sorted-runs](/compaction-num-sorted-runs.jpg)
+
+
+The number of sorted runs indicates the orderliness of SST files. More sorted runs typically lead to worse query performance. The main goal of compactions is to reduce the number of sorted runs.
+
+### Levels
+
+Databases based on LSM trees may also have levels, keys are merged level by level. GreptimeDB only has 2 levels which are 0 (uncompacted) and 1 (compacted).
+
+## Compaction strategies
+GreptimeDB provides two compaction strategies as mentioned above, but only time windowed compaction strategy (TWCS) can be chosen when creating tables. Strict window is only available when executing manual compactions.
+
+## Time windowed compaction strategy
+
+TWCS primarily aims to reduce read/write amplification during compaction.
+
+It assigns files to be compacted into different time windows. For each window, TWCS identifies the sorted runs. If the number of sorted runs exceeds the maximum allowed, TWCS finds a solution to reduce them to the threshold while considering merging penalties. If the number of sorted runs does not exceed the threshold, TWCS checks for excessive file fragmentation and merges fragmented files if necessary since SST file count also impacts query performance.
+
+For window assignment, SST files may span multiple time windows. TWCS assigns SSTs based on their maximum timestamps to ensure they are not affected by stale data. In time-series workloads, out-of-order writes are infrequent, and even when they occur, recent data's query performance is more critical than that of stale data.
+
+
+TWCS provides 4 parameters:
+- `max_active_window_runs`: max allowed sorted runs in the active window
+- `max_active_window_files`: max allowed files in the active window
+- `max_inactive_window_runs`: max allowed sorted runs in inactive windows
+- `max_inactive_window_files`: max allowed files in inactive windows
+
+Users can set different thresholds for active and inactive windows. This is important because out-of-order writes typically occur in the active window. By allowing more overlapping files in the active window, TWCS reduces write amplification during ingestion and merges all these files when the active window becomes inactive.
```

Comment:
```suggestion
You can set different thresholds for active and inactive windows. This is important because out-of-order writes typically occur in the active window. By allowing more overlapping files in the active window, TWCS reduces write amplification during ingestion and merges all these files when the active window becomes inactive.
```

Reply:
fixed in 4fdcf4d3